### PR TITLE
feat(Sync Logs): unify debug-sync logs between platforms.

### PR DESCRIPTION
**Pull request by asafkorem**
This pull request addresses the issue described here: Unify debug-sync logs between platforms #2714

As part of this change:

1. iOS: Update DetoxSync to the latest commit.
2. Android: Replace the old string sync-status with a JSON description
3. Unify debug-sync logs between the platforms, using a JSON formatter.

### Examples for _Android_ sync-status formatted messages:
```
detox[9160] INFO:  [APP_STATUS] The app seems to be idle
```
```
detox[9160] INFO:  [APP_STATUS] The app is busy with the following tasks:
• "LooperIdlingResource-2215-mqt_native_modules" (Native Modules Thread) is executing (native module calls).
• "LooperIdlingResource-2214-mqt_js" (JS Thread) is executing (JavaScript code).
• UI elements are busy:
  - Reason: UI rendering activity.
```

### Examples for _iOS_ sync-status formatted messages:

```
detox[9733] INFO:  [APP_STATUS] The app is busy with the following tasks:
• Run loop "JS Run Loop" is awake.
• There are 2 work items pending on the dispatch queue: "Main Queue (<OS_dispatch_queue_main: com.apple.main-thread>)".
• The event "Runloop Perform Block" is taking place with object: "JS Run Loop".
• The event "Runloop Perform Block" is taking place with object: "JS Run Loop".
• Run loop "Main Run Loop" is awake.
• The event "React Native (bundle load)" is taking place.
```

```
detox[9733] INFO:  [APP_STATUS] The app is busy with the following tasks:
• There are 1 work items pending on the dispatch queue: "Main Queue (<OS_dispatch_queue_main: com.apple.main-thread>)".
• Run loop "Main Run Loop" is awake.
• 1 enqueued native timers:
  - Timer #1:
    + Fire date: 2021-11-11 14:19:57 +0200.
    + Time until fire: 0.072.
    + Repeat interval: 0.
    + Is recurring: NO.
```

### CODE REVIEW

**Comment by asafkorem** [comment]
retest this please

**Comment by asafkorem** [comment]
@jonathanmos @d4vidi pushed native tests, integration tests are still missing. 
However, this should not block the reviews.

File: detox/android/detox/src/testFull/java/com/wix/detox/reactnative/idlingresources/AsyncStorageIdlingResourceSpec.kt [hunk 1]
```diff
@@ -70,6 +70,7 @@ class AsyncStorageIdlingResourceSpec: Spek({
                 givenAnActiveTask()
                 givenNoPendingTasks()
                 assertThat(uut.isIdleNow).isFalse()
+                assertThat(uut.getJSONDescription()).isEqualTo(mapOf("name" to "io"))
```
Comment (d4vidi) [inline detox/android/detox/src/testFull/java/com/wix/detox/reactnative/idlingresources/AsyncStorageIdlingResourceSpec.kt:73]: +1 point, Level Up

File: detox/android/detox/src/testFull/java/com/wix/detox/adapters/server/DetoxActionHandlersSpec.kt [hunk 2]
```diff
@@ -94,6 +96,61 @@ object DetoxActionHandlersSpec : Spek({
...
+                abstract val thread: String?
+            }
+
+            fun createMockedLooperIdlingResource(resourceName: String): LooperIdlingResource {
+                val resource: LooperIdlingResource = mock()
+                whenever(resource.name).thenReturn(resourceName)
+                return resource
+            }
+
+            it("should return busy app status") {
```
Comment (d4vidi) [inline detox/android/detox/src/testFull/java/com/wix/detox/adapters/server/DetoxActionHandlersSpec.kt:130]: split 3-ways

File: detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt [hunk 3]
```diff
@@ -100,30 +100,102 @@ class QueryStatusActionHandler(
...
+    }
+
+    /**
+     * @see URL https://reactnative.dev/docs/profiling
+     */
+    private fun isJSCodeExecution(looperName: String): Boolean {
+        val options = arrayOf(
+                "mqt_js",
+                "JSCall",
+                "Bridge.executeJSCall"
```
Comment (d4vidi) [inline detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt:163]: not needed

File: detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt [hunk 4]
```diff
@@ -100,30 +100,102 @@ class QueryStatusActionHandler(
...
+
+    /**
+     * @see URL https://reactnative.dev/docs/profiling
+     */
+    private fun isNativeCodeExecution(looperName: String): Boolean {
+        val options = arrayOf(
+                "mqt_native",
+                "NativeCall",
+                "callJavaModuleMethod",
+                "onBatchComplete"
```
Comment (d4vidi) [inline detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt:194]: not needed

File: detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt [hunk 5]
```diff
@@ -100,30 +100,102 @@ class QueryStatusActionHandler(
...
+                        if (executionType != null) {
+                            mapOf<String, Any>(
+                                    "thread" to thread,
+                                    "execution_type" to executionType
+                            )
+                        } else {
+                            mapOf<String, Any>(
+                                    "thread" to thread
+                            )
+                        }
```
Comment (d4vidi) [inline detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt:182]:
```suggestion
                          immutableMapOf<String, Any>("thread" to thread)
                            .let {
                              if (executionType != null) {
                                it["execution_type"] = executionType
                              } 
                            }
```

File: detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt [hunk 6]
```diff
@@ -100,30 +100,102 @@ class QueryStatusActionHandler(
     : DetoxActionHandler {
```
Comment (d4vidi) [inline detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt:100]: Extract class?

**Review by d4vidi — CHANGES_REQUESTED** [review]
1. Align documentation: debug sync guide (https://wix.github.io/Detox/docs/troubleshooting/synchronization/)
1. Inline comments

File: detox/android/detox/src/full/java/com/wix/detox/reactnative/idlingresources/BridgeIdlingResource.java [hunk 7]
```diff
@@ -42,8 +46,15 @@ public String getName() {
 
     @NotNull
     @Override
-    public String getDescription() {
-        return "Activity on the React-Native bridge";
+    public Map<String, Object> getJSONDescription() {
+        final Map<String, Object> jsonDescription = new HashMap<>();
+        jsonDescription.put(IdlingResourceConstants.RESOURCE_NAME_KEY, "one_time_event");
```
[the remark this answers is not in the document]
Comment (asafkorem) [reply detox/android/detox/src/full/java/com/wix/detox/reactnative/idlingresources/BridgeIdlingResource.java:51]: done :+1:

File: detox/android/detox/src/full/java/com/wix/detox/reactnative/idlingresources/BridgeIdlingResource.java [hunk 8]
```diff
@@ -42,8 +46,15 @@ public String getName() {
...
+
+        final Map<String, Object> description = new HashMap<>();
+        description.put("event", "Activity on the React-Native bridge");
+        jsonDescription.put(IdlingResourceConstants.RESOURCE_DESCRIPTION_KEY, description);
+
+        return jsonDescription;
```
[the remark this answers is not in the document]
Comment (asafkorem) [reply detox/android/detox/src/full/java/com/wix/detox/reactnative/idlingresources/BridgeIdlingResource.java:57]: done

File: detox/android/detox/src/testFull/java/com/wix/detox/adapters/server/DetoxActionHandlersSpec.kt (hunk 2 above)
Comment (asafkorem) [reply detox/android/detox/src/testFull/java/com/wix/detox/adapters/server/DetoxActionHandlersSpec.kt:130]: done

File: detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt (hunk 3 above)
Comment (asafkorem) [reply detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt:163]: 💯  good catch

File: detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt (hunk 5 above)
Comment (asafkorem) [reply detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt:182]: `.apply {` seems to fit better here

File: detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt (hunk 6 above)
Comment (asafkorem) [reply detox/android/detox/src/full/java/com/wix/detox/adapters/server/DetoxActionHandlers.kt:100]: done

**Comment by asafkorem** [comment]
Fixed and squashed.

**Pull request merged**